File: xa-api-php-sdk/xa/http/RequestParams.php
<?php
include_once '../model/Action.php';
include_once '../model/Update.php';
include_once '../model/Operation.php';

class RequestParams {
	
	var $requestParams;
	
	public function __construct($params) {
		$this->requestParams = array ();
		if (is_array ( $params )) {
			$operations = $params;
		} else {
			$operations = array (
					$params 
			);
		}
		$count = 0;
		foreach ( $operations as $operation ) {
			if (($operation instanceof Action) || ($operation instanceof Update)) {
				$this->requestParams [$operation->getOperationName ( $count ++ )] = $operation->getOperationValue ();
			}
		}
	}
	
	public function addParam($key, $value) {
		$this->requestParams [$key] = $value;
	}
	
	public function removeParam($key) {
		unset ( $this->requestParams [$key] );
	}
	
	public function toQueryString() {
		$first = true;
		foreach ( $this->requestParams as $key => $value ) {
			if (! $first) {
				$queryString .= '&';
			}
			$queryString .= $key . '=' . $value;
			$first = false;
		}
		return $queryString;
	}
}

?>

File: xa-api-python-sdk/xa/http/RequestParams.py
from xa.model.Operation import Operation


class RequestParams:

    def __init__(self, params):
        """

        :param params:
        """
        self.params = {}
        if isinstance(params,list):
            operations = params
        else:
            operations = [params]
            
        count = 0
        for operation in operations:
            if isinstance(operation, Operation):
                key = operation.getOperationName(count)
                value = operation.getOperationValue();
                count += 1
                self.params[key] = value

    def addParam(self, key, value):
        self.params[key] = value

    def removeParam(self, key):
        del self.params[key]

    def toQueryString(self):
        first = True
        queryString = ''
        for key in self.params.keys():
            if not first:
                queryString += '&'
            queryString += key + '=' + self.params[key]
            first = False
        return queryString
            
        
                
            
